src/domain/common/usecase/DecryptUrlUseCase.ts

```diff
@@ -0,0 +1,38 @@
+import EthCrypto from 'eth-crypto';
+
+import InvalidEncryptionError from '@core/domain/common/utils/error/InvalidEncryptionError';
+
+class DecryptUrlUseCase {
+  private typeRegex = new RegExp(/(encrypted\(.*?\))/g);
+
+  constructor(private readonly privateKey: string) {
+    this.privateKey = privateKey;
+  }
+
+  async decrypt(url: string): Promise<string> {
+    let result = url;
```

Comment:
you don't have to, strings are immutable in js

Reply:
What do you mean? `result` is supposed to be a modified version of `url`, in further processing I am recreating `result` based on its current value. I am not expecting to work on a reference of `url` there if that's what you mean (String is immutable in Java as well)

Reply:
Agreed face2face.